sdk/translation/azure-ai-translation-document/tests/test_all_document_statuses.py

```diff
@@ -18,65 +19,30 @@ class TestAllDocumentStatuses(DocumentTranslationTest):
-        blob_data = [Document(data=b'This is some text')]
-        source_container_sas_url = self.create_source_container(data=blob_data)
-        target_container_sas_url = self.create_target_container()
+        docs_count = 5
         target_language = "es"
 
-        # prepare translation inputs
-        translation_inputs = [
-            DocumentTranslationInput(
-                source_url=source_container_sas_url,
-                targets=[
-                    TranslationTarget(
-                        target_url=target_container_sas_url,
-                        language_code=target_language
-                    )
-                ]
-            )
-        ]
-
         # submit and validate job
-        job_id = self._submit_and_validate_translation_job(client, translation_inputs, len(blob_data))
+        job_id = self._create_translation_job_with_dummy_docs(client, docs_count, language_code=target_language, wait=False)
 
         # check doc statuses
         doc_statuses = list(client.list_all_document_statuses(job_id)) # convert from generic iterator to list
-        self.assertEqual(len(doc_statuses), len(blob_data))
+        self.assertEqual(len(doc_statuses), docs_count)
 
         for document in doc_statuses:
             self._validate_doc_status(document, target_language)
 
 
-    @pytest.mark.skip("top not exposed yet")
     @DocumentTranslationPreparer()
     @DocumentTranslationClientPreparer()
     def test_list_statuses_with_pagination(self, client):
-        # prepare containers and test data
-        blob_text = b'blob text'
-        blob_data = [Document(data=blob_text), Document(data=blob_text), Document(data=blob_text),
-                     Document(data=blob_text), Document(data=blob_text), Document(data=blob_text)]
-        source_container_sas_url = self.create_source_container(data=blob_data)
-        target_container_sas_url = self.create_target_container()
+        docs_count = 10
         result_per_page = 2
-        no_of_pages = len(blob_data) // result_per_page
+        no_of_pages = docs_count // result_per_page
         target_language = "es"
 
-        # prepare translation inputs
-        translation_inputs = [
-            DocumentTranslationInput(
-                source_url=source_container_sas_url,
-                targets=[
-                    TranslationTarget(
-                        target_url=target_container_sas_url,
-                        language_code=target_language
-                    )
-                ]
-            )
-        ]
-
         # submit and validate job
-        job_id = self._submit_and_validate_translation_job(client, translation_inputs, len(blob_data))
+        job_id = self._create_translation_job_with_dummy_docs(client, docs_count, language_code=target_language, wait=False)
 
         # check doc statuses
         doc_statuses_pages = list(client.list_all_document_statuses(job_id=job_id, results_per_page=result_per_page))
```

Comment:
If we're testing `results_per_page` here we should not convert to list, but rather call `by_page` on the result
```suggestion
        doc_statuses_pages = client.list_all_document_statuses(job_id=job_id, results_per_page=result_per_page).by_page()
```